### PR TITLE
Feat/add type lint

### PR DESCRIPTION
**Pull request by thisisarjun**
### Overview

This PR introduces major improvements to code quality, type safety, developer workflow, and test reliability for the MeetingMuse project.

#### 1. Type Annotations & Structure
- **Comprehensive type annotations** have been added throughout the codebase, covering function signatures, class attributes, and method arguments/returns.
- Refactored node and service classes for explicit attribute typing, improved constructor clarity, and enhanced static analysis compatibility.

#### 2. Node Class Refactor & Logic Improvements
- All node classes (e.g., `CollectingInfoNode`, `ClassifyIntentNode`, `PromptMissingMeetingDetailsNode`, etc.) now feature:
  - Explicit attribute typing.
  - Improved handling of message content types (e.g., safely handling both string and complex content).
  - Robust error handling (with `pylint` disables for broad exception catches where necessary).
- **Meeting details extraction and update logic** is more reliable and easier to unit test.
- **Router logic** is improved for type safety and clarity.

#### 3. Linting & Static Analysis
- **Integrated pre-commit hooks** for `black`, `isort`, `flake8`, `autoflake`, `mypy`, and standard pre-commit checks:
  - See changes in `.pre-commit-config.yaml`, `.flake8`, and related files.
  - Updated `Makefile` with formatting/linting commands.
- **Configuration updates** for `pyproject.toml`:
  - Added `autoflake` dependency.
  - Updated `pylint`, `mypy`, and other tool settings.
  - Added `[tool.pylint.*]` configuration blocks for consistent linting rules.

#### 4. Tests & Fixtures
- **Test suite refactored** for clarity, parameterization, and type consistency.
- Enhanced test coverage for edge cases in meeting detail updates, intent classification, and utility functions.

#### 5. Utilities & Logging
- Improved logger for type safety and better coloring support.
- Utilities are now more robust for message extraction, event handling, and state mutation.

#### 6. General Code Quality
- Consistent import ordering, whitespace, and formatting throughout.
- Reduced ambiguity in function/method responsibilities.
- Eliminated unused imports and variables.

---

### Breaking Changes

- Some method signatures and class constructors have changed to require explicit types.
- Pre-commit workflow is now mandatory for contributors.

**Please let me know if you have feedback or if any additional explanations are needed!**

### CODE REVIEW

**Comment by thisisarjun** [comment]
@ismailmmd the number of files are pretty large( 49) , but its mostly types and linter fixes, let me know if you need help reviewing

File: pyproject.toml [hunk 1]
```diff
@@ -17,6 +17,7 @@ dependencies = [
     "langchain-huggingface (>=0.3.0,<0.4.0)",
     "dotenv (>=0.9.9,<0.10.0)",
     "langchain-core (>=0.3.68,<0.4.0)",
+    "autoflake (>=2.3.1,<3.0.0)",
```
Comment (ismailmmd) [inline pyproject.toml:20]: Lets move this to dev dependencies

**Review by ismailmmd — COMMENTED** [review]
Otherwise looks okay

File: pyproject.toml (hunk 1 above)
Comment (thisisarjun) [reply pyproject.toml]: done. please review once more.

**Review by ismailmmd — APPROVED** [review]
